### PR TITLE
Fix test, add "--classmap-authoritative" to all composer install commands

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | `--classmap-authoritative` does two good things - it improves autoload speed and CI tests then if files are not excluded from autoload by `exclude-from-classmap` option like here: https://github.com/symfony/symfony/blob/v3.4.43/src/Symfony/Component/Validator/composer.json#L61 , see full trace below for a realworld example.
| Type?             | bug fix
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | n/a
| How to test?      | GH CI must pass
| Possible impacts? | no

```
Fatal error: Uncaught Error: Class 'Symfony\Component\Validator\Tests\Constraints\AbstractConstraintValidatorTest' not found in /builds/mvorisek/prestashop_eshop/tests-legacy/Unit/PrestaShopBundle/PassVsprintfContraintTest.php:35
Stack trace:
#0 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Util/FileLoader.php(57): include_once()
#1 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Util/FileLoader.php(45): PHPUnit\Util\FileLoader::load('/builds/mvorise...')
#2 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Framework/TestSuite.php(540): PHPUnit\Util\FileLoader::checkAndLoad('/builds/mvorise...')
#3 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Framework/TestSuite.php(618): PHPUnit\Framework\TestSuite->addTestFile('/builds/mvorise...')
#4 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Util/Configuration.php(1137): PHPUnit\Framework\TestSuite->addTestFiles(Array)
#5 /builds/mvorisek/prestashop_eshop/vendor/phpunit/phpunit/src/Util/Configuration.php(1003): P in /builds/mvorisek/prestashop_eshop/tests-legacy/Unit/PrestaShopBundle/PassVsprintfContraintTest.php on line 35
```

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23964)

### CODE REVIEW

File: tests-legacy/Unit/PrestaShopBundle/PassVsprintfContraintTest.php [hunk 1]
```diff
@@ -29,10 +29,10 @@
 use PrestaShopBundle\Entity\Translation;
 use PrestaShopBundle\Translation\Constraints\PassVsprintf;
 use PrestaShopBundle\Translation\Constraints\PassVsprintfValidator;
-use Symfony\Component\Validator\Tests\Constraints\AbstractConstraintValidatorTest;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
 use Symfony\Component\Validator\Validation;
 
-class PassVsprintfContraintTest extends AbstractConstraintValidatorTest
+class PassVsprintfContraintTest extends ConstraintValidatorTestCase
```
Comment (mvorisek) [inline tests-legacy/Unit/PrestaShopBundle/PassVsprintfContraintTest.php:35]: FYI: https://github.com/symfony/symfony/blob/v3.4.43/src/Symfony/Component/Validator/Tests/Constraints/AbstractConstraintValidatorTest.php#L17

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 install: composer assets
 
 composer:
-	composer install
+	composer install --ansi --prefer-dist --no-interaction --no-progress --classmap-authoritative
```
Comment (PierreRambaud) [inline Makefile:4]: It is maybe ok for CI, but I'm afraid we can't do it here:
https://github.com/PrestaShop/PrestaShop/pull/22146
https://github.com/PrestaShop/PrestaShop/issues/22141 :sweat_smile: 
We also need to keep the progress and remove ansi for CI.

Comment (mvorisek) [reply Makefile:4]: I addressed that in https://github.com/PrestaShop/PrestaShop/pull/23982

About ansi/progress, I did not introduce it, see https://github.com/PrestaShop/PrestaShop/pull/23964/files#diff-a73bb6555480a5ee79ae276a3f5d71a08fa316e09a4a8da7b643cf1e92c97df9R29 , I only normalized it across other uses. Please why you do not want that and if that can stay.

**Comment by PierreRambaud** [comment]
We will not introduce `--classmap-authoritative` because of these: #22146
#22141

**Comment by mvorisek** [comment]
@PierreRambaud cherry pick the test fix at least

**Comment by mvorisek** [comment]
@PierreRambaud this is for CI and with this change, the CI will be stricter, what is wrong with that?

In `Makefile`, which can be used outside CI, I can revert this change. Ok with you then?

**Comment by PierreRambaud** [comment]
It not represent the way PrestaShop is provide, we have to keep the same behavior between CI and zip file for testing purpose.

**Comment by mvorisek** [comment]
@PierreRambaud but this proposed CI update is strictly stronger, thus CI can only find hidden issues, releases can continue using `composer` without this change

**Pull request closed without merging**